src/gt4py/next/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -181,9 +181,11 @@ def _preprocess_program(
         self,
         program: itir.FencilDefinition,
         offset_provider: dict[str, Connectivity | Dimension],
-    ) -> itir.FencilDefinition | global_tmps.FencilWithTemporaries | itir.Program:
+    ) -> itir.Program:
         if not self.enable_itir_transforms:
-            return program
+            return fencil_to_program.FencilToProgram().apply(
+                program
+            )  # TODO(tehrengruber): should be removed after refactoring to combined IR
```

Comment:
Does this `TODO` imply that it should be tracked in issue #1582 as another cleanup task?